pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -0,0 +1,77 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	compute "google.golang.org/api/compute/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
+	"k8s.io/klog/v2"
+)
+
+// PerformMultiNetworkCIDRAllocation allots pod CIDRs for all the networks that a node is connected to. It handles IPv6 only for default-network for now.
+func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, interfaces []*compute.NetworkInterface) (defaultNwCIDRs []string, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation, err error) {
+	k8sNetworksList, err := ca.networkClient.NetworkingV1().Networks().List(context.TODO(), metav1.ListOptions{})
```

Comment:
It will be important to evaluate the cost of API calls in this method is called in every nodeUpdate IIUIC , and it performs a List here (Get apiserver/) and another one below `GKENetworkParamSets()` ... but not for this PR

Reply:
I don't really know how watches and informers work for CRDs honestly

Reply:
it seems it can https://github.com/kubernetes/cloud-provider-gcp/pull/417, can we add an infomer and use the Lister instead of doing direct calls?

Reply:
Ack, thanks for suggesting this.